File: ECommerceApp-Backend-master/CartService/src/main/java/com/csi/service/CouponService.java
package com.csi.service;

import com.csi.exception.CouponNotFoundException;
import com.csi.model.Coupon;
import org.springframework.cache.annotation.Cacheable;

import java.util.List;

public interface CouponService {
    Coupon createCoupon(Coupon coupon);

    @Cacheable(value = "couponId")
    Coupon getCoupon(String couponId);

    Coupon updateCoupon(Coupon coupon);

    void deleteCoupon(String couponId);

    List<Coupon> getAllCoupon();

    Coupon getByName(String couponName);
}

File: ECommerceApp-Frontend-main/src/services/CouponService.js
import axios from "axios"
import { cartService } from "../constants"

class CouponService {

    createCoupon(coupon) {
        return axios.post(cartService.COUPON + '/create', coupon)
    }

    getCoupon(couponId) {
        return axios.get(cartService.COUPON + `/get/${couponId}`)
    }

    updateCoupon(couponId, coupon) {
        return axios.put(cartService.COUPON + `/update/${couponId}`, coupon)
    }

    deleteCoupon(couponId) {
        return axios.delete(cartService.COUPON + `/delete/${couponId}`)
    }

    getAllCoupons() {
        return axios.get(cartService.COUPON + '/')
    }

}
export default new CouponService()